frontend/src/routes/decrypt/+page.svelte

```diff
@@ -11,92 +12,173 @@
   import { onMount } from "svelte";
   import Auth from "./auth.svelte";
   import Password from "./password.svelte";
+  import { getError, hasError } from "$lib/form";
+  import { AlertCircle } from "lucide-svelte";
 
   onMount(() => {
     renderLoginButton(document.getElementById("login-button"));
   });
 
-  let encodedPayload = "";
-  let plainText: Promise<string> | null = null;
-  let error: any = null;
-  let decrypt: () => Promise<string>;
+  const DecryptFormSchema = z.object({
+    encodedPayload: z.string().min(1),
```

Comment:
For encoded payload, here's what I want for the validation logic (it's basically on line 55 below, but with some additional logic):
1. Try to `Cipherly.decodePayload(formData.encodedPayload)` - if you get Base64 decode error, validation should say, "Invalid Cipherly payload"
2. If you successfully decode base64, check to see if there's an `es` field, if not, "Invalid Cipherly payload (missing encryption scheme)"
3. If the `es` field is `Password`, ensure that the `s`, `iv`, and `ct` fields are present, if not "Invalid Cipherly payload (missing salt / iv / ciphertext)"
4. If the `es` field is `Auth`, ensure ... you get the point.

Reply:
That makes sense, I'll look into this.